src/convertdate/tibetan.py

```diff
@@ -0,0 +1,258 @@
+        self.md2 = Decimal(11135) / Decimal(11312)
+        self.md0 = 2015501 + Decimal(4783) / Decimal(5656)
+
+        self.s1 = Decimal(65) / Decimal(804)
+        self.s2 = Decimal(13) / Decimal(4824)
+        self.s0 = Decimal(743) / Decimal(804)
+
+        self.a1 = Decimal(253) / Decimal(3528)
+        self.a2 = Decimal(1) / Decimal(28)
+        #self.a2 = Decimal(3781) / Decimal(105840)
+        self.a0 = Decimal(475) / Decimal(3528)
+
+        self.moon_tab = [0, 5, 10, 15, 19, 22, 24, 25]
+        self.sun_tab = [0, 6, 10, 11]
+
+sch = Cal_school(y0 = 806, m0 = 3, beta_star = 61) # phugpa
+
+def true_month_leap(y: int, m: int) -> Tuple[int, bool]:
+    '''Calculate a true month count of year/month
+       Also return whether it is leap month or not
+    '''
+    solar_month_cnt: int = 12 * (y - sch.y0) + m - sch.m0
+    true_month_rcnt: float = (67 * solar_month_cnt + sch.beta_star) / 65
+    ix: int = (67 * solar_month_cnt + sch.beta_star) % 65
+    if ix in [sch.ix_leap, sch.ix_leap+1]:
+        leap_month = True
+    else:
+        leap_month = False
+    
+    if ix < sch.ix_leap:
+        true_month_cnt = math.floor(true_month_rcnt)
+    elif ix in [sch.ix_leap, sch.ix_leap+1]:
+        true_month_cnt = math.floor(true_month_rcnt)
+    else:
+        true_month_cnt = math.ceil(true_month_rcnt)
+    return(true_month_cnt, leap_month)
+
+def mean_date(n: int, d: int):
+    return(n * sch.md1 + d * sch.md2 + sch.md0) 
+
+def mean_sun(n: int, d: int):
+    return(n * sch.s1 + d * sch.s2 + sch.s0)
+
+def anomaly_moon(n: int, d: int):
+    return(n * sch.a1 + d * sch.a2 + sch.a0)
+
+def moon_tab_int(i: int) -> int:
+    k = i % 28
+    if k <= 7:
+        return(sch.moon_tab[k])
+    elif k <= 14:
+        return(sch.moon_tab[14-k])
+    elif k <= 21:
+        return(-sch.moon_tab[k-14])
+    else:
+        return(-sch.moon_tab[28-k])
+
+def sun_tab_int(i: int):
+    k = i % 12
+    if k <= 3:
+        return(sch.sun_tab[k])
+    elif k <= 6:
+        return(sch.sun_tab[6-k])
+    elif k <= 9:
+        return(-sch.sun_tab[k-6])
+    else:
+        return(-sch.sun_tab[12-k])
+
+def tab_float(f: Callable, a: float) -> float:
+    a_floor: int = math.floor(a)
+    a_frac = a - a_floor
+    a_tab = f(a_floor)
+    a_tab_plus = f(a_floor+1)
+    a_tab_float: float = a_tab + (a_tab_plus - a_tab) * a_frac
+    return(a_tab_float)
+
+def true_date(n: int, d: int):
+    me = tab_float(moon_tab_int, 28 * anomaly_moon(n, d)) # moon_equ
+    se = tab_float(sun_tab_int, 12 * (mean_sun(n, d) - Decimal(0.25))) # sun_equ
+    td = mean_date(n, d) + Decimal(me / 60) - Decimal(se / 60)
+
+    return(td)
+
+def inc_lunar_day(n, d):
+    if d == 30:
+        n += 1
+        d = 1
+    else:
+        d += 1
+    return (n, d)
+
+def dec_lunar_day(n, d):
+    if d == 1:
+        n -= 1
+        d = 30
+    else:
+        d -= 1
+    return (n, d)
+
+def to_jd(year, month, leap_month, day, leap_day):
+    '''Obtain Julian day for Tibetan date'''
```

Comment:
Are the `leap_month` and `leap_day` arguments necessary here? Shouldn't the converter be able to check if the given month/day is leap?